### ISSUE TITLE
Why minSdkVersion is 15 for this library?

### ISSUE THREAD

**Issue by DASAR** [onepf/OPFIab#31]
A can't build my app which has minSdkVersion=14. There is any reason for minSdk 15 ?

**Comment by RomanZhilich** [comment]
Hello.

Well, basically SDK 14 was never popular. You can see here https://developer.android.com/about/dashboards/index.html that it has less then 0.1% share and it pretty much was like that since it's appearance.

**Comment by DASAR** [comment]
@RomanZhilich  so basicly I'm asking why not support Android 2.3? Any technical reason?

**Comment by Lakedaemon** [comment]
On 08/18/2015 11:21 AM, Bogdasarov Bogdan wrote:
>
> @RomanZhilich <https://github.com/RomanZhilich> so basicly I'm asking 
> why not support Android 2.3? Any technical reason?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/onepf/OPFIab/issues/31#issuecomment-132136225>.
>
Just in cse, the RxBinding people had just the same argument here : 
https://github.com/JakeWharton/RxBinding/issues/48

And one solution for the user was to override the minSdk of the library

One can just add the following to their manifest to force their SDK 
versions over the library:

<uses-sdk  tools:overrideLibrary="com.jakewharton.rxbinding"  />

**Comment by RomanZhilich** [comment]
> which has minSdkVersion=14
> why not support Android 2.3

SDK 14 is Android 4.0

As for Android 2.3... well during development there were some API used from newer versions.
Right now the only place I can remember is here https://github.com/onepf/OPFIab/blob/fb9eed1cc93b8a094d095afc2744de6d0731409c/opfiab/src/main/java/org/onepf/opfiab/OPFIab.java#L245

**Comment by RomanZhilich** [comment]
@Lakedaemon 
> One can just add the following to their manifest to force their SDK

This won't work for 2.3 due to activity lifecycle callback API I mentioned above.

**Comment by DASAR** [comment]
My application is API 14, Android 4.0. I was tried to override library version using this kind of directive with no luck.
```xml
<uses-sdk  tools:overrideLibrary="com.jakewharton.rxbinding"  />
```
Anyway thanks for responding. I got you.

**Comment by RomanZhilich** [comment]
> tried to override library version using this kind of directive with no luck.

Have you tried `<uses-sdk  tools:overrideLibrary="org.onepf.opfiab"  />`?

**Comment by DASAR** [comment]
@RomanZhilich yes. Later I found this format in StackOverflow:
```xml
<uses-sdk tools:overrideLibrary="com.example.lib1, com.example.lib2"/>
```
Assume this one will work. Because I have to apply override not only to "org.onepf.opfiab", but also for "org.onepf:opfutils" and other libraries (Google, Amazon, etc).

**Comment by mikemee** [comment]
In the fwiw department, according [to this](https://en.wikipedia.org/wiki/Fire_OS), the original Kindle Fire used API Level 10 (2.3.3), but everything else since has used API 15 (4.0.3) or higher (17 then 19).

I just checked my active installed base (past month), and I have about 1% of sessions on API < 15. A lot of that seems to be the first Kindle Fire (_though it's a little hard to be sure_).  I'm not sure if I'm ready to abandon them yet... 

@RomanZhilich will you accept a PR if I locate and tweak other possible places where newer API calls are used (i.e. in addition to the `Application.ActivityLifecycleCallbacks` you already referenced)?  Or is it not worth it, in your opinion (I may well be missing some nuances here)?  

_PS: many thanks for putting the effort into this updated library!_

**Comment by RomanZhilich** [comment]
> will you accept a PR

Well.. it sorta depends... You see, it's not like I don't value other peoples efforts, but it's kinda risky investment of time. There's a good chance I won't like something in the code, and we won't know that until you send your PR.

There's another good reason to not support earlier API's a soon as they become outdated. Android is quite buggy. There's all kinds of problems all over the core framework: Handlers, URLConnection, AsyncTask you name it.
Usually this problems can be worked around with ```if(Build.VERSION.SDK_INT == ...``` but it can become difficult to support over time. Not mentioning the temptation of using new shiny API features...

Bottom line - I don't think it's worth it to support API below 15. If 2.3.3 is not dead yet it will probably be soon enough.

**Comment by mikemee** [comment]
> Bottom line is - I don't think it's worth the effort to support API below 15. If 2.3.3 is not dead yet it will probably be soon enough.

Exactly the sort of guidance I was looking for, thanks! 

_I think I'll just drop support for those oldest tablets going forward and they can keep using the current version of the app._
